app/routes.py

```diff
@@ -0,0 +1,57 @@
+import re
+from app import app
+from flask import request, Response, redirect, abort
+from app.helpers import add_item, fetch_url
+from app.helpers.checks import check_params
+from app.helpers.response_generation import create_response
+from service_config import Config
+
+
+@app.route('/checker', ['GET'])
+def checker():
+    return 'OK', 200
+
+
+"""
+The create_shortlink route's goal is to take an url and create a shortlink to it.
+The only parameter we should receive is the url to be shortened.
+In the route, we check the url received for the following comportments : [SHORTENED_URL]
+if the url is not from an allowed host or domain, we refuse it. 
+https://github.com/geoadmin/mf-chsdi3/blob/a75335af46f156f7ad9c10efe17eb52c888d671b/chsdi/lib/helpers.py#L130
+we also check from who the query comes, and here is the expected comportment: [HOST]
+if it comes from the api, the redirection host becomes s.geo.admin.ch
+if it comes from somewhere else (for example: if we deploy the service in a test environment) : 
+https://github.com/geoadmin/mf-chsdi3/blob/a75335af46f156f7ad9c10efe17eb52c888d671b/chsdi/lib/helpers.py#L105
+host = request.host + ('' if [dev, int, prod] else '/user_name) if host is not Localhost, else 
+localhost
+if agnostic: return ''.join(('//', host) --> //boulgour.admin.ch
+else return ''.joins(request.scheme, '://', host) --> https://boulgour.admin.ch
+
+final result --> [HOST + SHORTENED_URL]
+"""
+
+
+@app.route('/shorten', ['GET'])
+@app.route('/shorten.json', ['GET'])
+def create_shortlink():
+    r = request
+    response_headers = {'Content-Type': 'application/json; charset=utf-8'}
+    url = r.args.get('url', None)
+    scheme = r.scheme
+    domain = r.url_root.replace(scheme, '')  # this will return the root url without the scheme
+    base_path = r.script_root
+    base_response_url = check_params(scheme, domain, url, base_path)
+    response = create_response(base_response_url, add_item(url))
+    if r.headers.get('Origin') is not None and \
+            re.match(Config.allowed_domains_pattern, request.headers['Origin']):
+        response_headers['Access-Control-Allow-Origin'] = r.headers['origin']
+        response_headers['Access-Control-Allow-Methods'] = 'GET, OPTION'
+        response_headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization,' \
+                                                           ' x-requested-with, Origin, Accept'
+    return response, 200, response_headers
```

Comment:
The main difference, for now, is that this mimics the current implementation, which means this would still work with the current calls from the current viewer. If we change the method, we would break things. 
I agree with you, but we need to make things clear: do we develop these only for the future viewer (in which case, this is a resource creation and we should go for a POST query), or do we want to stay compatible at least for now, and we keep the GET for now.

Reply:
From my understanding we should go for the new way of doing things with the drawback that new services will only run with new viewer, but @boecklic to confirm.

Reply:
jep, no need to support current behaviour, only new viewer will use this service so we can change to POST here as well.
This also means, we can change the routes, imho no need for `shorten.json` anymore, I'd just go for 
`POST /shorten`
`GET  /resolve/{hash}` or `GET /redirect` or sth alike (or even both? `GET /resolve/{hash}` is just returning the resolved URL in a json, `GET /redirect/{hash}` is returning a 301 to the new URL, might be handy for debugging? or overkill?)